imap-proto/src/core.rs

```diff
@@ -59,6 +59,12 @@ pub fn quoted_data(i: &[u8]) -> IResult<&[u8], &[u8]> {
     Ok((&i[len..], &i[..len]))
 }
 
+/// space = a single space character
+named!(pub space, tag_s!(" "));
+
+// nil = "NIL"
+named!(pub nil, tag_s!("NIL"));
```

Comment:
I'd be hesitant about this, too, but since it's in the syntax it might be helpful.